### PR TITLE
ROX-15355: Remove HTPasswd creation

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

This PR removes the creation of the HTPasswd provider, since all new DP clusters should make use of the OIDC provider.

It also changes the way one obtains a token to login to the cluster to create the service account used during provisioning the dataplane clusters:
`ocm cluster login` will be used and it's expected to interactively login through browser to obtain the token, and pass the token as input to the cluster.

Since the script currently is only executed manually and during setup of the dataplane cluster, this should be sufficient enough for the time being.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.

## Test manual

```
# Tested against a staging cluster (removing the part of re-creating the service account)
./osd-cluster-idp-setup.sh stage acs-stage-dp-02
```

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* main
  * **PR #947** <a href="https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/947" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/947?utm_source=stack-comment).

File: dp-terraform/osd-cluster-idp-setup.sh [hunk 1]
```diff
@@ -47,18 +46,6 @@ export_cluster_environment() {
     init_chamber
     load_external_config "osd" OSD_
     load_external_config "cluster-$CLUSTER_NAME" STORED_
```
Comment (vladbologa) [inline dp-terraform/osd-cluster-idp-setup.sh:48]: I think this line is no longer needed.

**Review by kovayur — CHANGES_REQUESTED** [review]
Thanks @dhaus67, just a small remark on the cleanup.

**Review by kovayur — APPROVED** [review]

**Pull request merged**